Assets/Mirror/Runtime/ClientScene.cs

```diff
@@ -127,24 +124,13 @@ public static bool Ready(NetworkConnection conn)
             if (conn != null)
             {
                 conn.Send(new ReadyMessage());
-                ready = true;
-                readyConnection = conn;
-                readyConnection.isReady = true;
+                conn.isReady = true;
                 return true;
             }
             Debug.LogError("Ready() called with invalid connection object: conn=null");
             return false;
         }
 
-        internal static void HandleClientDisconnect(NetworkConnection conn)
-        {
-            if (readyConnection == conn && ready)
-            {
-                ready = false;
-                readyConnection = null;
-            }
-        }
-
```

Comment:
can we keep HandleClientDisconnect first and simply do the variable changes?
one small step after another :)